src/main/java/com/internet/shop/dao/ProductDaoImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        return Storage.addProduct(product);
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        return Storage.products.stream()
+                .filter(product -> product.getId() == id)
+                .findFirst();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        return Storage.products.stream()
+                .filter(productFromStorage -> productFromStorage.getId() == (product.getId()))
+                .limit(1)
+                .peek(x -> x.setPrice(product.getPrice()))
+                .peek(x -> x.setName(product.getName()))
```

Comment:
what if you will ad other fields to Product?

Reply:
fixed it